### DISCUSSION TITLE
Use builtins or not?

### DISCUSSION THREAD

**Discussion by lukeanavarro** [codewars/codewars.com#2643] [General]
Is the intent of this site to code solutions from scratch or to code using the languages full set of tools. 

For example:
Find the maximum in a list could be (in python)
max = 0 
for I in list: 
if I > max: 
max = i 
return max

or just return max(list)

On the one hand if you are just trying to master a language then the latter is best but if you are trying to learn how to solve problems the former might be better.

Thoughts?

**Comment by iorchalvarez** [comment]
The short answer would be both.

The idea is for you to solve the problem according to your knowledge, but at the same time watch how other people are solving the same problem. This will give you more ideas. Sometimes, specially when you're learning, you only know certain way to solve it, maybe it will take you longer to do it, but after watching the other solutions, you will find better and/or easier ways to do it.

**Comment by SDEllis1978** [comment]
So, maybe a bit late to this party but saw it and wanted to add:

I have seen more than my fair share of exercises over time that will explicitly state not to use a builtin tool, or conversely will link towards a doc page of a builtin tool or even a some place teaching multiple ways of doing the same thing.   

That being said, over the years, I've come to the conclusion that if you CAN use the builtin, use it.   If they don't want you to, they'll tell you, or prevent you from using it.

**Comment by karakarasuuuu** [comment]
It totally depends on what language you use and what you want to learn from Codewars.  
For example, I think the latter one is more pythonic but the former one can learn more. As long as you understand what you have done, it's up to you.
